Fix deprecated createFactory warning

w3bdesign opened the pull request:
Fixing issue https://github.com/frontity/frontity/issues/384

**What**:

Fix for issue https://github.com/frontity/frontity/issues/384.

As this is my first contribution, I am sorry if I am not following guidelines 100% or doing everything correctly. 
Please feel free to correct my contribution so that it follows contributing guidelines.

**Why**:

I see the following after cloning the mars starter theme and running the dev server:

Warning: React.createFactory() is deprecated and will be removed in a future major release. 
Consider using JSX or use React.createElement() directly instead.

**How**:

I replaced the Spinner component with a custom CSS animation using @emotion and keyframes.

For reference I used https://projects.lukehaas.me/css-loaders/ and adopted it to utilize @emotion.

**Tasks**:

- [x] Code
- [ ] TypeScript
- [ ] Unit tests
- [ ] End to end tests
- [ ] TypeScript tests
- [x] Update starter themes
- [ ] Update other packages
- [ ] Documentation
- [ ] Community discussions
- [ ] Changeset

** Screenshot**:

![image](https://user-images.githubusercontent.com/45217974/80656732-2c30d200-8a82-11ea-92ec-d86c0f6cfb00.png)

w3bdesign committed:
Fix deprecated createFactory warning

Fixing issue https://github.com/frontity/frontity/issues/384

w3bdesign committed:
Update -webkit-animation

Update -webkit-animation with loading variable

michalczaplinski commented:
Thanks for the contribution @w3bdesign ! 🙂 

The github action has failed, but it's not your fault, PRs from external collaborators cannot deploy previews to now.sh because they don't have access to our token. We're going to fix that with a bot in the next few days. 

In the meantime, someone from the team is gonna have a look at your PR!

michalczaplinski committed:
Merge branch 'dev' into dev

michalczaplinski commented:
@github-actions run

<details>
<summary>🚀 Merge preview</summary>

```js
// Get pull-req URL like "https://api.github.com/repos/nwtgck/actions-merge-preview/pulls/4"
const pullReqUrl = context.payload.issue.pull_request.url;
const githubUser = context.payload.repository.owner.login;
const res = await fetch(pullReqUrl, {
  headers: [
    ['Authorization', `Basic ${Buffer.from(`${githubUser}:${githubToken}`).toString('base64')}`]
  ]
});
const resJson = await res.json();
const prUserName = resJson.head.user.login;
const baseBranchName = resJson.base.ref;
const branchName = resJson.head.ref;
const fullRepoName = resJson.head.repo.full_name;
const previewBranchName = `actions-merge-preview/${prUserName}-${branchName}`;
execSync(`git config --global user.email "github-actions[bot]@users.noreply.github.com"`);
execSync(`git config --global user.name "github-actions[bot]"`);
// (from: https://stackoverflow.com/a/23987039/2885946)
execSync(`git fetch --all`);
console.log(execSync(`git checkout ${baseBranchName}`).toString());
console.log(execSync(`git checkout -b ${previewBranchName} ${baseBranchName}`).toString());
console.log(execSync(`git pull https://github.com/${fullRepoName}.git ${branchName}`).toString());
// Push preview branch
// NOTE: Force push (should be safe because preview branch always start with "actions-merge-preview/")
execSync(`git push -fu origin ${previewBranchName}`);
const baseRepoFullName = context.payload.repository.full_name;
// Comment body
const commentBody = `🚀 Preview branch:  \n<https://github.com/${baseRepoFullName}/tree/${previewBranchName}>`;
// Comment the deploy URL
await postComment(commentBody);
```
</details>

michalczaplinski commented on packages/mars-theme/src/components/spinner.js:
```diff
@@ -0,0 +1,54 @@
+  border-radius: 50%;
+  position: relative;
+  text-indent: -9999em;
+  -webkit-animation: ${loading} 1.3s infinite linear;
```
@w3bdesign You don't actually need to add any vendor prefixes for any CSS properties like `-webkit-something` or `-ms-something`. Emotion adds those prefixes automatically, so you can safely remove those! :)

w3bdesign replied:
@michalczaplinski See https://github.com/frontity/frontity/pull/409/commits/571c39446bc2eb7ccbec4aa35c7b9050ad5e6e34

michalczaplinski commented:
Looks great @w3bdesign ! 👍 Thanks for the contribution, I just have one small change request above.

And let us know if you'd like to make the same update to the twentytwenty theme 🙂 Otherwise, it's good to merge after the fix.

@DAreRodz any more comments?

michalczaplinski committed:
Merge branch 'dev' into dev

w3bdesign committed:
Add Spinner to Twentytwenty

Also remove vendor prefix

michalczaplinski commented:
Awesome @w3bdesign! 🙂 Thanks for the contribution! 

I will wait till Monday with merging it though 😄

michalczaplinski reviewed (approved):
Looks good, although we maybe want to adapt the design to keep the current style of 2020 theme loader? Any opinions?

w3bdesign commented:
> 
> 
> Looks good, although we maybe want to adapt the design to keep the current style of 2020 theme loader? Any opinions?

Let me know and I'll do a new commit.

luisherranz reviewed (changes requested):
Hey guys, awesome work 🙂 

Regarding the design, I'd prefer if we stick to the current loader, which I already created here: https://github.com/luisherranz/embed-files-in-node-server/blob/master/frontity/packages/mars-theme/src/components/loading.js

Apart from that, I think we don't need two files (`loading.js`, `spinner.js`) for this, only one is enough. This is the theme people use to learn Frontity so we need to make it as simple as possible.

luisherranz commented:
@w3bdesign in case you also want to help us with the TwentyTwenty loader, I took a look a while ago.

- This is the code: https://github.com/davidhu2000/react-spinners/blob/9a5836c545b9324df6ec5ae67fba39dff864fb7b/src/BarLoader.tsx
- And it has a lot of helpers, but I think the only one you really need to match the design is a function to add alpha to a color expressed in hex. It turns out it's easy to replicate:

```js
function calculateRgba(hex, alpha) {
  var r = parseInt(hex.slice(1, 3), 16),
    g = parseInt(hex.slice(3, 5), 16),
    b = parseInt(hex.slice(5, 7), 16);

  if (alpha) {
    return "rgba(" + r + ", " + g + ", " + b + ", " + alpha + ")";
  } else {
    return "rgb(" + r + ", " + g + ", " + b + ")";
  }
}

calculateRgba("#FF0000", 0.5);
```

https://stackoverflow.com/a/28056903/1830574

w3bdesign committed:
Revert "Add Spinner to Twentytwenty"

This reverts commit 571c39446bc2eb7ccbec4aa35c7b9050ad5e6e34.

w3bdesign committed:
Implement changes according to luisherranz

Also deleted Spinner.js

w3bdesign commented:
> 
> 
> Hey guys, awesome work slightly_smiling_face
> 
> Regarding the design, I'd prefer if we stick to the current loader, which I already created here: https://github.com/luisherranz/embed-files-in-node-server/blob/master/frontity/packages/mars-theme/src/components/loading.js
> 
> Apart from that, I think we don't need two files (`loading.js`, `spinner.js`) for this, only one is enough. This is the theme people use to learn Frontity so we need to make it as simple as possible.

I reverted the changes to TwentyTwenty for now and implemented the changes you mentioned to the Mars theme, can you approve and merge it for now?

w3bdesign reviewed (commented):
See the latest commits.

w3bdesign committed:
Add changeset on behalf of w3bdesign

luisherranz reviewed (approved):
Awesome. Many thanks for the contribution @w3bdesign! 🎉
